src/routes/v1/trip-patterns.ts

```diff
@@ -175,9 +183,58 @@ router.post<'/replace-tripPattern', { id: string }>(
                     return leg
                 })
 
+                // If first Leg is foot we have to do some manual modification to the Tripattern
+                let expectedStartTime = ''
+                if (newLegs[0]?.id === null && newLegs[1]?.expectedStartTime) {
+                    expectedStartTime = toISOString(
+                        subSeconds(
+                            parseISO(newLegs[1]?.expectedStartTime),
```

Comment:
Denne trenger vell ikke null-check siden det blir gjort i `if`'en. Ser ut som man kan sende inn `undefined` til `parseISO` som kanskje vil krasje.
```suggestion
                            parseISO(newLegs[1].expectedStartTime),
```